engine/kv_engine.cpp

```diff
@@ -1344,123 +1353,133 @@ Status KVEngine::SGet(const StringView collection, const StringView user_key,
   return skiplist->Get(user_key, value);
 }
 
-Status KVEngine::GetTTL(const StringView str, TTLTimeType* ttl_time) {
+Status KVEngine::GetTTL(const StringView str, TTLType* ttl_time) {
   HashTable::KeyHashHint hint = hash_table_->GetHint(str);
   std::unique_lock<SpinMutex> ul(*hint.spin);
-  HashEntry hash_entry;
-  HashEntry* entry_ptr = nullptr;
-  Status s = hash_table_->SearchForRead(hint, str, ExpirableRecordType,
-                                        &entry_ptr, &hash_entry, nullptr);
-
-  ExpiredTimeType expired_time;
-  *ttl_time = kInvalidTTLTime;
-  if (s != Status::Ok) {
-    return s;
+  LookupResult res = lookupKey(str, ExpirableRecordType);
+  if (res.s == Status::Expired) {
+    // Push the expired record into cleaner and update hash entry status with
+    // HashEntryStatus::Expired.
+    // TODO(zhichen): This `if` will be removed when completing collection
+    // deletion.
+    if (res.entry_ptr->GetIndexType() == HashIndexType::StringRecord) {
+      hash_table_->UpdateEntryStatus(res.entry_ptr, HashEntryStatus::Expired);
+      delayFree(OldDeleteRecord{res.entry_ptr->GetIndex().ptr,
```

Comment:
Don't delayFree here. If GetTTL() delayFree() an expired record, other API also should do this.